roundrobin_weighted.go

```diff
@@ -80,9 +80,10 @@ func (w *RRW) Next() interface{} {
 			w.cw = w.cw - w.gcd
 			if w.cw <= 0 {
 				w.cw = w.maxW
-				if w.cw == 0 {
-					return nil
-				}
+				// When does this happen?
```

Comment:
see http://kb.linuxvirtualserver.org/wiki/Weighted_Round-Robin_Scheduling, the clang implementation.

Reply:
I has create a issue talk about those code. https://github.com/smallnest/weighted/issues/3 
but now, revert that code in this commit.